### ISSUE TITLE
A better returning value for draw_networkx_edges

### ISSUE THREAD

**Issue by sanghack81** [networkx/networkx#3833]
`draw_networkx_edges` returns either `LineCollection` or `arrow_collection` (which is a list of `FancyArrowPatch`) depending on whether the given graph is `DiGraph` and `arrows=True`.

The documentation says
```
matplotlib.collection.LineCollection
    `LineCollection` of the edges

list of matplotlib.patches.FancyArrowPatch
    `FancyArrowPatch` instances of the directed edges
```

Internally, the function is short-circuited by checking whether the `edgelist` is empty:
```python
if not edgelist or len(edgelist) == 0:  # no edges!
    return None
```

Wouldn't it be better if 
```python
if not edgelist or len(edgelist) == 0:  # no edges!
    if G.is_directed() and arrows:
        return []
    else:
        return LineCollection(None)
```

Currently, the documentation does not mention that it returns `None` if no edge is drawn. 

p.s. `draw_networkx_labels` returns an empty set if nothing is drawn. `draw_networkx_nodes` returns `None` but it may be able to return `PathCollection(None)` since documentation did not specify its behavior. 

p.p.s. I am using `zip(edgelist, arcs)` after `arcs = draw_networkx_edges...`. It would be better to have an empty list than `None` which needs an additional check.

**Comment by dschult** [comment]
This seems like a reasonable request. Let's make the return values match the container type that is described in the doc_string. If no edges/nodes are drawn, then return a container with no edges/nodes.

**Comment by rossbar** [comment]
It looks like this was closed by #3854. Please reopen if I've missed something.
